src/object/reader.rs

```diff
@@ -186,6 +190,56 @@ impl AsyncSeek for ObjectReader {
     }
 }
 
+impl tokio::io::AsyncRead for ObjectReader {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut tokio::io::ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        let b = buf.initialize_unfilled();
+        let n = ready!(futures::AsyncRead::poll_read(self.as_mut(), cx, b))?;
```

Comment:
Please use `self.inner.poll_read()` to use the `output:Read`'s implementation instead go thougth `futures` implementation.